SETUP.md

```diff
@@ -1,330 +1,406 @@
 
-```shell
-bazel build //cmd/triage:triage
-./bazel-bin/triage/triage_/triage -c backend-config.yaml
-```
+## Optional Features
 
-#### Setting up Yara scanning
-
-You can optionally setup automatic yara scanning of the malware that is
-collected. To do this, you need to run this command from the code root
-directory, where your backend-config.yaml is:
-```shell
-go run cmd/yara/main.go -m -r <path-to-yara-rules> -b 100
+### VirusTotal Integration
+Add to `.env.backend`:
+```env
+LOPHIID_VIRUSTOTAL_API_KEY=your-virustotal-api-key
 ```
 
-We use Yara-X so some older yara rules might potentially not work or
-need updating. The code simply ignores those rules (but does output an error on
-the screen).
-
-This process will regularly poll the database for new downloads and then runs
-the loaded rules against them. The results are stored in the database and
-visible in the UI (e.g. downloads tab).
-
-# Setting up the agent
-
-## Prepare privilege dropping
-
-Create the chroot directory and create a user to drop privileges to.
+### Telegram Alerting
+Add to `.env.backend`:
+```env
+LOPHIID_ALERTING_TELEGRAM_API_KEY=your-bot-token
+LOPHIID_ALERTING_TELEGRAM_CHAT_ID=your-chat-id
+```
 
-```shell
-mkdir -p -m 755 /var/empty/lophiid
-useradd -d /var/empty/lophiid -M -r -s /bin/false lophiid-agent
+### LLM Integration
+Add to `.env.backend`:
+```env
+LOPHIID_AI_TRIAGE_ENABLE=true
+LOPHIID_AI_API_LOCATION=http://localhost:8000/v1
+LOPHIID_AI_API_KEY=your-api-key
+LOPHIID_AI_MODEL=your-model-name
 ```
 
-Next update the configuration and set 'chroot_dir' to the chroot directory
-and 'user' to the user created with the command above.
+## Advanced Usage
 
-## Allow ICMP packets
+### Multiple Agents per Machine
+```bash
+# Generate multiple agent configs with unique names
+./scripts/setup-agent-certs.sh <backend-ip> <agent-ip> agent-1
+./scripts/setup-agent-certs.sh <backend-ip> <agent-ip> agent-2
 
-Use the following command to allow ICMP ping by the agent. The example is a bit
-lazy and allows it to all users. You can pudate the group range with a group ID
-that lophiid-agent belongs to in order to make it more strict.
-```shell
-sudo sysctl -w net.ipv4.ping_group_range="0 2147483647"
+# Each gets unique auth token and certificate
+# Deploy each agent separately with different port mappings
+# Register each agent separately in the UI with unique tokens
 ```
 
-## Create the configuration
-
-The configuration options are documented in the [config file](./config/agent-config.yaml).
-Note that you need to run one agent per IP so you might have to run multiple
-agents on a single machine and each is configured individually.
+### Scaling Multiple Agents
+When deploying multiple agents across different machines:
+1. Use the same CA certificates (agents are network agnostic)
+2. Generate unique auth tokens for each agent (64 hex characters)
+3. Register each agent separately in the backend UI
+4. Consider using different port ranges to avoid conflicts
```

Comment:
I'm not sure I understand why.   Maybe we can simplify the documentation and just mention that we support multiple agents per machine as long as each agent has it's dedicated public IP.  Multiple agents can listen on the same IP address but in that case the ports each agent listens on cannot overlap.

Before this change though agents could not listen on the same IP because the IP was a unique differentiator in the backend. I'm not sure if you change this but I assume you did. If not than the suggestion above needs to be ignored and the documentation needs updating